### PR TITLE
Add IngestAuthAgent and include authentication headers

### PR DESCRIPTION
**Pull request by rexwangcc**
In this PR:
1. I add an IngestAuthAgent class to interact with the new Ingest Service.
2. Include authentication headers where it's necessary for the test suite.

### CODE REVIEW

File: tests/ingest_agents.py [hunk 1]
```diff
@@ -72,9 +79,69 @@ def status(self):
...
             url = self.ingest_api_url + f'/submissionEnvelopes/{self.envelope_id}'
-            self.data = requests.get(url).json()
+            self.data = requests.get(url, headers=self.auth_headers).json()
+
+
+class IngestAuthAgent:
+    def __init__(self,
+                 url="https://danielvaughan.eu.auth0.com/oauth/token",
+                 client_id="Zdsog4nDAnhQ99yiKwMQWAPc2qUDlR99",
+                 client_secret="t-OAE-GQk_nZZtWn-QQezJxDsLXmU7VSzlAh9cKW5vb87i90qlXGTvVNAjfT9weF",
```
[the remark this answers is not in the document]
Comment (sampierson) [reply tests/ingest_agents.py]: Instructions are here https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml
Please add a comment like before each encrypted environment variable in travis.yml as the encrypted line also encrypts the variable name.
Oh and @danielvaughan needs to generate new ones now that these are compromised.

**Comment by sampierson** [comment]
Please merge this PR to master, not staging.  Master has the  code and is the base for the deployment specific branches.  After it is merged into master, merge master into staging.

File: tests/ingest_agents.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sampierson) [reply tests/ingest_agents.py]: Okay I guess you can ignore this.  I read the comment in _get_auth_token().
You might move that comment closer to the embedded creds.

[the remark this answers is not in the document]
Comment (rexwangcc) [reply tests/ingest_agents.py]: Thanks for the comment, I want to post Daniel's comment from another PR here:
> The secret gives you the ability to request a token. However, anyone with a Google account can also get a token. At the moment there is no authorisation and the token is purely used to enable us to identify authenticated users and fill in owner field for the submission envelope in the database. The token that the secret gives you is a catch-all as it allows us to require the endpoint is sent and token and allows us to put something in the database for submission created without logging in via google. We also use the same token for creating submission through the previous UI. No one can do anything with the secret that they could do not do without. The secret can also revoked at any time should that change.

And yeah, I should have moved the docstring closer to the creds, will do that right away.

**Comment by rexwangcc** [comment]
Change base from `staging` to `master` branch.

**Pull request merged**